src/lib/y2partitioner/actions/controllers/disk_device.rb

```diff
@@ -0,0 +1,196 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+Yast.import "Mode"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # This class stores needed information to perform certain actions over a disk device,
+      # for example, the disks over which to clone the device. It also takes care of
+      # updating the devicegraph when needed.
+      class DiskDevice
+        # Current disk device
+        #
+        # @return [Y2Storage::BlkDevice] it should be a disk device,
+        #   (i.e., device#is?(:disk_device) #=> true).
+        attr_reader :device
+
+        # @return [Array<Y2Storage::Partitionable>]
+        attr_accessor :selected_devices_for_cloning
+
+        # Constructor
+        #
+        # @param device [Y2Storage::BlkDevice] it should be a disk device,
+        #   (i.e., device#is?(:disk_device) #=> true).
+        def initialize(device)
+          @device = device
```

Comment:
please verify it is really disk_device otherwise we get nasty failures later

Reply:
Ok.